pkg/util/vmopv1/vm.go

```diff
@@ -213,3 +216,123 @@ func IsClasslessVM(vm vmopv1.VirtualMachine) bool {
 func IsImagelessVM(vm vmopv1.VirtualMachine) bool {
 	return vm.Spec.Image == nil && vm.Spec.ImageName == ""
 }
+
+func SyncStorageUsageForNamespace(
+	ctx context.Context,
+	k8sClient client.Client,
+	namespace, storageClass string) (bool, <-chan error) {
+
+	var (
+		chanErr = make(chan error, 1)
+		objKey  = client.ObjectKey{
+			Namespace: namespace,
+			Name:      spqutil.StoragePolicyUsageName(storageClass),
+		}
+	)
+
+	syncFn := func() {
+		defer close(chanErr)
+
+		ctx := context.Background()
+
+		// Make sure the StoragePolicyUsage document exists.
+		var obj spqv1.StoragePolicyUsage
+		if err := k8sClient.Get(ctx, objKey, &obj); err != nil {
+
+			// Even if the SPU is not found, log the error to indicate that
+			// the function was enqueued but could not proceed.
+			chanErr <- fmt.Errorf(
+				"failed to get StoragePolicyUsage %s: %w", objKey, err)
+			return
+		}
+
+		var list vmopv1.VirtualMachineList
+		if err := k8sClient.List(
+			ctx,
+			&list,
+			client.InNamespace(namespace)); err != nil {
+
+			chanErr <- fmt.Errorf(
+				"failed to list VMs in namespace %s: %w", namespace, err)
+			return
+		}
+
+		var (
+			totalUsed     resource.Quantity
+			totalReserved resource.Quantity
+		)
+
+		for i := range list.Items {
+			vm := list.Items[i]
+
+			if !vm.DeletionTimestamp.IsZero() || vm.Status.Storage == nil {
+				// Ignore VMs being deleted or that are not reporting storage
+				// status.
+				continue
+			}
+
+			// Account for storage used by this VM.
+			if vm.Status.Storage.Unshared != nil {
+				totalUsed.Add(*vm.Status.Storage.Unshared)
```

Comment:
@SandeepPissay / @xing-yang, question for y'all,

To calculate the total space that could be used by a VM (i.e. reserved), we should be summing Committed and Uncommitted from https://dp-downloads.broadcom.com/api-content/apis/API_VWSA_001/8.0U3/html/ReferenceGuides/vim.vm.Summary.StorageSummary.html, right?